### ISSUE TITLE
stacklevel used by `deprecate_func` seems wrong

### ISSUE THREAD

**Issue by lagru** [scikit-image/scikit-image#7568]
Currently working on https://github.com/scikit-image/scikit-image/pull/7566, and it looks like using a function decorated with `deprecate_func` raises a `FutureWarning` that points at https://github.com/scikit-image/scikit-image/blob/8e37f77568b580be56424adc3ac4b5515da3ab2e/skimage/_shared/utils.py#L537

instead of the actual location where the deprecated function is used. Not a priority right now, but should be investigated.

**Comment by stefanv** [comment]
I thought at some point we figured out how to calculate the stacklevel automatically?

**Comment by lagru** [comment]
Yes, thought so to. At least we are now catching these in places where we are using `assert_stacklevel`.
